### PR TITLE
src, test: Adds a basic `FindInputs` action

### PR DESCRIPTION
**Pull request by surovic**
This action heuristically detects inputs to tools and records them after the tool is run. The tool works much like the existing `FindOutputs` action.

### CODE REVIEW

**Comment by woodruffw** [comment]
LGTM, one small nit.

File: src/blight/actions/find_inputs.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+class FindInputs(Action):
+    def before_run(self, tool: Tool) -> None:
+        inputs = []
+        for input in tool.inputs:
+            input_path = Path(input)
+            if not input_path.is_absolute():
+                input_path = tool.cwd / input_path
+
+            kind = INPUT_SUFFIX_KIND_MAP.get(input_path.suffix, InputKind.Unknown)
```
Comment (pgoodman) [inline src/blight/actions/find_inputs.py:90]: Should probably do `input_path.suffix.lower()` to handle things like `.C`, etc.

Comment (woodruffw) [reply src/blight/actions/find_inputs.py:90]: Yeah, let's do a followup for that. I bet `FindOutputs` is also probably wrong.

Comment (pgoodman) [reply src/blight/actions/find_inputs.py:90]: I have some similar code to this in PASTA. Here's what I do:

```c++
    std::vector<std::string> new_argv;
    for (const char *arg : job_args) {

      // Try to fixup any remaining paths.
      llvm::StringRef a(arg);

      // Try to look for things that look file file names, then see if they are
      // file names, and if so, make them absolute paths.
      if (a.contains("./") || a.endswith_insensitive(".o") ||
          a.endswith_insensitive(".cc") || a.endswith_insensitive(".hh") ||
          a.endswith_insensitive(".cpp") || a.endswith_insensitive(".hpp") ||
          a.endswith_insensitive(".c++") || a.endswith_insensitive(".h++") ||
          a.endswith_insensitive(".cxx") || a.endswith_insensitive(".hxx") ||
          a.endswith_insensitive(".c") || a.endswith_insensitive(".h") ||
          a.endswith_insensitive(".gcno") || a.endswith_insensitive(".pch") ||
          a.endswith_insensitive(".s") || a.endswith_insensitive(".asm") ||
          a.endswith_insensitive(".mm") || a.endswith_insensitive(".d") ||
          a.endswith_insensitive(".sdk")) {

        if (auto maybe_info = fs.Stat(arg); maybe_info.Succeeded()) {
          new_argv.emplace_back(
              maybe_info.TakeValue().real_path.generic_string());
          continue;
        }
      }

      new_argv.emplace_back(arg);
    }
```

Comment (pgoodman) [reply src/blight/actions/find_inputs.py:90]: `-include foo.h`.

Comment (woodruffw) [reply src/blight/actions/find_inputs.py:90]: 😭

**Pull request merged**